Sources/web3swift/Utils/EIP/EIP712.swift

```diff
@@ -48,14 +47,15 @@ public extension EIP712Hashable {
             case let data as EIP712.Bytes:
                 result = data.sha3(.keccak256)
             case is EIP712.UInt8:
-                result = ABIEncoder.encodeSingleType(type: .uint(bits: 8), value: field as AnyObject)!
+                result = ABIEncoder.encodeSingleType(type: .uint(bits: 8), value: field)!
             case is EIP712.UInt256:
-                result = ABIEncoder.encodeSingleType(type: .uint(bits: 256), value: field as AnyObject)!
+                result = ABIEncoder.encodeSingleType(type: .uint(bits: 256), value: field)!
             case is EIP712.Address:
-                result = ABIEncoder.encodeSingleType(type: .address, value: field as AnyObject)!
+                result = ABIEncoder.encodeSingleType(type: .address, value: field)!
             case let hashable as EIP712Hashable:
                 result = try hashable.hash()
             default:
+                /// Cast to `AnyObject` is required. Otherwise, `nil` value will fail this condition.
                 if (field as AnyObject) is NSNull {
```

Comment:
This is the only case found so far where `AnyObject` is required.
`field` value can be `nil` even though the type it has is `Any` (interesting edge case).
So checking for `field == nil` will return `false` if `field` is actually `nil`.